### PR TITLE
Allow miq_token as an argument passed into the ManageIQ API Client

### PR DESCRIPTION
**Pull request by syncrou**
##### SUMMARY
ManageIQ is an open source management platform for Hybrid IT.

This change is modifying:
```
manageiq.py to allow passing in a manageiq api token for use 
in authenticating with the manageiq-api-client-python
```

##### ISSUE TYPE
```
 Feature Pull Request
```

##### COMPONENT NAME
manageiq.py (module utils and doc_fragement)

##### ANSIBLE VERSION
```
ansible 2.3.0.0
```

### CODE REVIEW

**Comment by syncrou** [comment]
-cc @gmcculloug

**Comment by jctanner** [comment]
cc @gtanzillo @abellotti @zgalor @yaacov

**Comment by yaacov** [comment]
you also need to update the docs:
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/manageiq.py

File: lib/ansible/module_utils/manageiq.py
Comment (yaacov) [inline lib/ansible/module_utils/manageiq.py]: are out sure you want to return here ?

**Comment by yaacov** [comment]
@syncrou ping ^^

**Comment by syncrou** [comment]
@yaacov, yep, definitely found an issue. I have the fix in place, working at updating the docs now. Planning to push the changes today. Thanks for your review!

**Review by yaacov — APPROVED** [review]
LGTM 👍 

p.s.
I think you can remove the final `else` in `validate_connection_params` and un-indent the `for args in ...` loop, but it's a matter of taste :-)

**Comment by yaacov** [comment]
nice work with the docs,

some nitpicks :-)

I think this will be nicer:
```
if (url and username and password) or (url and token):
       return params

for arg in ['url', 'username', 'password']:
   if params[arg] in (None, ''):
       module.fail_json(msg=error_str.format(arg))
```

a. you do not need the double check for missing 'url' [ the second elif will catch it ]
b. the `for args` is `else` anyway , so we do not need to explicitly say `else:`

**Comment by syncrou** [comment]
@yaacov - Definitely a good point on the double checking for `url`.  I'll get those changes pushed.

**Comment by yaacov** [comment]
@syncrou the way I understand it ( i'm new here too :-) ) 

If you think this is ready, you should:

a. you need to ask for a spot in the next core IRC meeting here:
https://github.com/ansible/community/issues/220
b. you need to attend the next core meeting:
https://github.com/ansible/community/tree/master/meetings

**Comment by syncrou** [comment]
@yaacov - Ok - I wasn't sure, since this is an update to an existing module ( mod_utils ) 

I'll see what I need to do to follow up on the meeting.

**Comment by yaacov** [comment]
> Ok - I wasn't sure, since this is an update to an existing module ( mod_utils )

I'm new here too ... but I do not know of another way to get a review from someone with merge rights ...

cc @jctanner ?

**Comment by syncrou** [comment]
@yaacov - When you submitted for the `manageiq_user` module to be merged - were you in the Core meeting? Also, it looks like there will be no meetings next week due to the summit, so either way looks like I just missed it.

**Comment by yaacov** [comment]
> were you in the Core meeting?

yes, I was, and @cben was in another, it took more then one meeitng ... :-)

> it looks like there will be no meetings next week due to the summit,

then just wait, at least you know what is going on.

**Comment by jctanner** [comment]
bot_status

**Comment by jctanner** [comment]
Hi everyone. Just wanted to clarify a bit on what's going on here. The PR got a "ship it" from @yaacov  22 hours ago in a review comment. Unfortunately the bot does not look at review comments for "ship it"s yet, and those need to be in regular comments. There was also a commit made after @yaacov's review, so any "ship it"s that were made up till that time were reset.

So what needs to happen now ...
@yaacov or any other maintainer needs to add a "ship it" (no spaces) in a normal comment and if nobody else does, the core team can give it the final vote to get it merged.

**Comment by yaacov** [comment]
shipit

**Comment by syncrou** [comment]
@jctanner I noticed that although Shippable is PROCESSING it seems there is a failed test around spacing in `manageiq.py`. If I push again to fix that test will we need another 'shipit' from a maintainer?

**Comment by yaacov** [comment]
> will we need another 'shipit' from a maintainer?

np, I'll put another one :-)

**Comment by syncrou** [comment]
Thanks @yaacov - I'll push a fix.

**Comment by jctanner** [comment]
@syncrou yep, new commits will reset the counter every time as a way to ensure the maintainers have approved the change.

**Comment by yaacov** [comment]
I will wait for it to be green and re-send the "ship it"

**Comment by syncrou** [comment]
@yaacov - Cool - It's all green

**Comment by yaacov** [comment]
shipit

**Pull request merged**
